apps/sms/services/js/worker_client.js

```diff
@@ -0,0 +1,215 @@
+/*global bridge,
+ ClientDisposer,
+ disposableClient,
+ streamClient
+ */
+
+(function(exports) {
+  'use strict';
+
+  const debug = 1 ?
+    (arg1, ...args) => console.log(`[WorkerClient] ${arg1}`, ...args):
+    () => {};
+
+  /**
+   * Supported worker types.
+   * @enum {string}
+   */
+  const WORKER_TYPE = Object.freeze({
+    WORKER: 'worker',
+    SHARED_WORKER: 'shared-worker'
+  });
+
+  /**
+   * Supported lifetime strategies or the way we manage workers.
+   * @enum {string}
+   */
+  const LIFETIME_STRATEGY = Object.freeze({
```

Comment:
I left this ```enum``` from my previous WIP where I have 3rd strategy - ```PER_CALL``` when client is created for method call/stream and automatically disposed once method is finished/stream is closed. But I removed this for now - so can remove this enum as well if you wish.